pipes-and-filters/PipesAndFilters.Shared/ServiceBusPipeFilter.cs

```diff
@@ -37,56 +46,52 @@ public void Start()
+            this.sender = this.queue.CreateSender(this.outQueuePath);
         }
 
-        public void OnPipeFilterMessageAsync(Func<BrokeredMessage, Task<BrokeredMessage>> asyncFilterTask, int maxConcurrentCalls = 1)
+        public void OnPipeFilterMessageAsync(Func<ServiceBusReceivedMessage, Task<ServiceBusMessage>> asyncFilterTask)
         {
-            var options = new OnMessageOptions()
-            {
-                AutoComplete = true,
-                MaxConcurrentCalls = maxConcurrentCalls
-            };
-
-            options.ExceptionReceived += this.OptionsOnExceptionReceived;
-
-            this.inQueue.OnMessageAsync(
-                async (msg) =>
+            this.processor.ProcessMessageAsync +=
+                async args =>
             {
+                ServiceBusReceivedMessage message = args.Message;
                 pauseProcessingEvent.WaitOne();
 
                 // Perform a simple check to dead letter potential poison messages.
                 //  If we have dequeued the message more than the max count we can assume the message is poison and deadletter it.
-                if (msg.DeliveryCount > Constants.MaxServiceBusDeliveryCount)
+                if (message.DeliveryCount > Constants.MaxServiceBusDeliveryCount)
                 {
-                    await msg.DeadLetterAsync();
+                    await this.receiver.DeadLetterMessageAsync(message);
 
-                    Trace.TraceWarning("Maximum Message Count Exceeded: {0} for MessageID: {1} ", Constants.MaxServiceBusDeliveryCount, msg.MessageId);
+                    Trace.TraceWarning("Maximum Message Count Exceeded: {0} for MessageID: {1} ", Constants.MaxServiceBusDeliveryCount, message.MessageId);
 
                     return;
                 }
 
                 // Process the filter and send the output to the next queue in the pipeline
-                var outMessage = await asyncFilterTask(msg);
+                var outMessage = await asyncFilterTask(message);
 
                 // Send the message from the filter processor to the next queue in the pipeline
-                if (outQueue != null)
+                if (queue != null)
                 {
-                    await outQueue.SendAsync(outMessage);
+                    await this.sender.SendMessageAsync(outMessage);
                 }
 
                 //// Note: There is a chance we could send the same message twice or that a message may be processed by an upstream or downstream filter at the same time.
                 ////       This would happen in a situation where we completed processing of a message, sent it to the next pipe/queue, and then failed to Complete it when using PeakLock
                 ////       Idempotent message processing and concurrency should be considered in the implementation.
-            },
-            options);
+            };
+            processor.ProcessErrorAsync += this.OptionsOnExceptionReceived;
```

Comment:
We should try to be consistent with using "this". I see that the existing code uses it for class members, so we should continue to do that, unless we want to remove all instances of it. I'm fine with either approach but let's be consistent.